### ISSUE TITLE
Add limit_stacksize param to InvRef:add_item

### ISSUE THREAD

**Issue by SmallJoker** [minetest/minetest#9097]
##### Minetest version
```
5.1.0+
```

##### Summary
I'd like to multiply node drops which does not lead into oversized stacks.
```Lua
local handle_node_drops = minetest.handle_node_drops
-- Function to override several node definitions
	local old_def = minetest.registered_nodes[node_name]
	local drop_stack = old_def.drop and ItemStack(old_def.drop)
		or ItemStack(node_name)
	local drop_count = drop_stack:get_count()

-- minetest.override_node()
		stack_size = 2,
		after_dig_node = function(pos, oldnode, oldmetadata, digger)
			if math.random() > 0.9 then
				-- Multiply drops
				drop_stack:set_count(drop_count * 10)
				handle_node_drops(pos, {drop_stack}, digger)
			end
		end,
```
But `minetest.handle_node_drops` uses `add_item(` which does not handle oversized stacks properly.

##### Steps to reproduce
1) Override a node using the snippet above
2) Dig that node
3) Stack with size of 10, or 20 in your inventory

Marked as controversial because there are surely mods that use `add_item` to exactly have oversized stacks.

**Comment by SmallJoker** [comment]
Caused by #8519 which reverted #5774, which itself fixed #5773 which is in conflict to #8518.

A proper solution is needed here.

**Comment by SmallJoker** [comment]
Lost/no interest.
